File: src/main/java/infinitespire/relics/crystals/WardingShard.java
package infinitespire.relics.crystals;

import com.megacrit.cardcrawl.actions.common.ApplyPowerAction;
import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
import com.megacrit.cardcrawl.powers.DexterityPower;
import infinitespire.InfiniteSpire;
import infinitespire.abstracts.InfiniteRelic;

public class WardingShard extends InfiniteRelic {

    public static final String ID = InfiniteSpire.createID("WardingShard");

    public WardingShard(){
        super(ID, "wardingshard", LandingSound.CLINK);
    }

    @Override
    public void atBattleStart() {
        AbstractDungeon.actionManager.addToBottom(
            new ApplyPowerAction(
                AbstractDungeon.player,
                AbstractDungeon.player,
                new DexterityPower(AbstractDungeon.player, counter), counter));
    }
}

File: src/main/kotlin/com/blanktheevil/infinitespire/relics/WardingShard.kt
package com.blanktheevil.infinitespire.relics

import com.blanktheevil.infinitespire.extensions.applyPower
import com.blanktheevil.infinitespire.extensions.makeID
import com.blanktheevil.infinitespire.extensions.player
import com.blanktheevil.infinitespire.relics.abstracts.CrystalRelic
import com.megacrit.cardcrawl.powers.DexterityPower

class WardingShard : CrystalRelic(ID, IMG) {
  companion object {
    val ID = "WardingShard".makeID()
    private const val IMG = "wardingshard"
  }

  override fun atBattleStart() {
    player.applyPower(
      DexterityPower(player, counter),
      amount = counter
    )
  }
}